From 7996b58419d44bfaf99a4c2e0d0210d72cc4da40 Mon Sep 17 00:00:00 2001
From: Simon Mikuda <simon.mikuda@streamunlimited.com>
Date: Fri, 7 Feb 2020 13:01:25 +0100
Subject: [PATCH] common: main.c: make show_boot_progress __weak

This is upstream commit 3422299dc28fa8257677d03cc1253e3c9bf17e9f

This not only looks a bit better it also prevents a
warning with W=1 (no previous prototype).

Signed-off-by: Jeroen Hofstee <jeroen@myspectrum.nl>
Acked-by: Simon Glass <sjg@chromium.org>
---
 common/main.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/common/main.c b/common/main.c
index 8b6f274..be89af2 100644
--- a/common/main.c
+++ b/common/main.c
@@ -27,8 +27,7 @@ DECLARE_GLOBAL_DATA_PTR;
 /*
  * Board-specific Platform code can reimplement show_boot_progress () if needed
  */
-void inline __show_boot_progress (int val) {}
-void show_boot_progress (int val) __attribute__((weak, alias("__show_boot_progress")));
+__weak void show_boot_progress(int val) {}
 
 #define MAX_DELAY_STOP_STR 32
 
-- 
2.7.4

